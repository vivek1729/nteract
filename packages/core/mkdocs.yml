site_name: nteract/core

nav:
  - Overview: "overview.md"
