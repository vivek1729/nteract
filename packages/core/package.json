{
  "name": "@nteract/core",
  "version": "15.0.0",
  "description": "core modules and components for nteract apps",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "nteractDesktop": "src/index.ts",
  "scripts": {},
  "dependencies": {
    "@nteract/actions": "^7.0.0",
    "@nteract/commutable": "^7.3.0",
    "@nteract/epics": "^5.0.0",
    "@nteract/reducers": "^5.0.0",
    "@nteract/selectors": "^3.0.0",
    "@nteract/types": "^7.0.0",
    "redux-logger": "^3.0.6"
  },
  "peerDependencies": {
    "immutable": "^4.0.0-rc.12"
  },
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "nteract",
    "redux"
  ],
  "author": "Kyle Kelley <rgbkrk@gmail.com>",
  "license": "BSD-3-Clause"
}
