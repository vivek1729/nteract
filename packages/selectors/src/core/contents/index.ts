import { AppState, ContentRef, KernelRef } from "@nteract/types";

/**
 * Returns the contents, such as notebooks and files, that are currently accessible
 * within the current notebook application.
 *
 * @param   state   The state of the nteract application
 *
 * @returns         The contents in scope by the nteract application by ID
 */
export const contentByRef = (state: AppState) =>
  state.core.entities.contents.byRef;

/**
 *
 * @param   state                       The state of the nteract application
 * @param   { contentRef: ContentRef }  A reference to the ContentRecord to retrieve
 *
 * @returns                             The ContentRecord for the given ref
 */
export const content = (
  state: AppState,
  { contentRef }: { contentRef: ContentRef }
) => contentByRef(state).get(contentRef);

/**
 * Returns the model within the ContenteRecrd specified by contentRef.
 * For example, if the ContentRecord is a reference to a notebook object,
 * the model would contain the NotebookModel.
 *
 * @param   state                       The state of the nteract application
 * @param   { contentRef: ContentRef }  A reference to the ContentRecord to retrieve
 *
 * @returns                             The model of the content under the current ref
 */
export const model = (
  state: AppState,
  { contentRef }: { contentRef: ContentRef }
) => {
  const content = contentByRef(state).get(contentRef);
  if (!content) {
    return null;
  }
  return content.model;
};

export const notebookModel = (
  state: AppState,
  { contentRef }: { contentRef: ContentRef }
) => {
  const notebook = model(state, { contentRef });

  if (!notebook || notebook.type !== "notebook") {
    throw new Error("Expected notebook model");
  }

  return notebook;
};

/**
 * Returns a ref to the kernel associated with a particular type of content.
 * Currently, this only support kernels associated with notebook contents.
 * Returns null if there are no contents or if the contents are not
 * a notebook.
 *
 * @param   state                       The state of the nteract application
 * @param   { contentRef: ContentRef }  A reference to the ContentRecord to retrieve
 *
 * @returns                             The kernel associated with a notebook
 */
export const kernelRefByContentRef = (
  state: AppState,
  ownProps: { contentRef: ContentRef }
): KernelRef | null | undefined => {
  const c = content(state, ownProps);
  if (c && c.model && c.model.type === "notebook") {
    return c.model.kernelRef;
  }

  return null;
};

/*
 * Returns the kernel associated with the current ContentRef.
 */
export const kernelByContentRef = (
  state: AppState,
  ownProps: { contentRef: ContentRef }
) => {
  const kernelRef = kernelRefByContentRef(state, ownProps);
  if (kernelRef) {
    return state.core.entities.kernels.byRef.get(kernelRef);
  }

  return null;
};

/**
 * Returns the filepath for the content identified by a given ContentRef.
 *
 * @param   state     The state of the nteract application
 * @param   ownProps  An object containing the ContentRef
 *
 * @returns           The filepath for the content
 */
export const filepath = (
  state: AppState,
  ownProps: { contentRef: ContentRef }
): string | null => {
  const c = content(state, ownProps);
  if (!c) {
    return null;
  }
  return c.filepath;
};

/**
 * Returns the ContentRef associated with a given filepath.
 *
 * @param   state       The state of the nteract application
 * @param   ownProps    An object containing the filepath
 *
 * @returns             The ContentRef for the content under a filepath
 */
export const contentRefByFilepath = (
  state: AppState,
  ownProps: { filepath: string }
): string | undefined => {
  const byRef = contentByRef(state);
  return byRef.findKey(content => content.filepath === ownProps.filepath);
};
