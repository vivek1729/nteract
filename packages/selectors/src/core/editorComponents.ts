import { AppState } from "@nteract/types";

export const editorComponentsbyEditorType = (state: AppState) =>
  state.core.entities.editorComponents.byEditorType;

export const editorComponent = (state: AppState, { id }: { id: string }) =>
  editorComponentsbyEditorType(state).get(id);
