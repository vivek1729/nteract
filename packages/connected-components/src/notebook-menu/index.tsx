// Vendor modules
import { CellType } from "@nteract/commutable";
import { actions } from "@nteract/core";
import {
  AppState,
  ContentRef,
  HostRecord,
  KernelRef,
  KernelspecsByRefRecordProps,
  KernelspecsRef
} from "@nteract/types";
import { RecordOf } from "immutable";
import { connect } from "react-redux";
import { Dispatch } from "redux";

// Local modules
import { MODAL_TYPES } from "../modal-controller";
import PureNotebookMenu from "./PureNotebookMenu";

export function makeMapStateToProps(
  initialState: AppState,
  initialProps: { contentRef: ContentRef }
) {
  const { contentRef } = initialProps;

  const currentContentRef = contentRef;

  const mapStateToProps = (state: AppState) => {
    const content = state.core.entities.contents.byRef.get(contentRef);
    const host: HostRecord = state.app.host;
    const isBookstoreEnabled: boolean = host.bookstoreEnabled || false;

    // The current kernelspecs setup is _overkill_ as we're only ever going to
    // have one collection of kernelspecs

    const currentKernelspecsRef: KernelspecsRef | null =
      state.core.currentKernelspecsRef || null;
    const currentKernelspecs: RecordOf<
      KernelspecsByRefRecordProps
    > | null = currentKernelspecsRef
      ? state.core.entities.kernelspecs.byRef.get(currentKernelspecsRef, null)
      : null;

    // This menu should only work for notebooks
    if (!content || content.type !== "notebook") {
      return {
        currentContentRef,
        currentKernelRef: null,
        currentKernelspecs,
        currentKernelspecsRef
      };
    }

    const currentKernelRef = content.model.kernelRef;

    return {
      bookstoreEnabled: isBookstoreEnabled,
      currentContentRef,
      currentKernelRef,
      currentKernelspecs,
      currentKernelspecsRef
    };
  };

  return mapStateToProps;
}

export function makeMapDispatchToProps(
  initialDispatch: Dispatch,
  initialProps: { contentRef: ContentRef }
) {
  const mapDispatchToProps = (dispatch: Dispatch) => ({
    onPublish: (payload: { contentRef: string }) =>
      dispatch(actions.publishToBookstore(payload)),
    toggleNotebookHeaderEditor: (payload: { contentRef: string }) =>
      dispatch(actions.toggleHeaderEditor(payload)),
    saveNotebook: (payload: { contentRef: string }) =>
      dispatch(actions.save(payload)),
    downloadNotebook: (payload: { contentRef: string }) =>
      dispatch(actions.downloadContent(payload)),
    executeCell: (payload: { id: string; contentRef: string }) =>
      dispatch(actions.executeCell(payload)),
    executeAllCells: (payload: { contentRef: string }) =>
      dispatch(actions.executeAllCells(payload)),
    executeAllCellsBelow: (payload: { contentRef: string }) =>
      dispatch(actions.executeAllCellsBelow(payload)),
    clearAllOutputs: (payload: { contentRef: string }) =>
      dispatch(actions.clearAllOutputs(payload)),
    unhideAll: (payload: {
      outputHidden: boolean;
      inputHidden: boolean;
      contentRef: string;
    }) => dispatch(actions.unhideAll(payload)),
    cutCell: (payload: { id?: string; contentRef: string }) =>
      dispatch(actions.cutCell(payload)),
    copyCell: (payload: { id?: string; contentRef: string }) =>
      dispatch(actions.copyCell(payload)),
    pasteCell: (payload: { contentRef: string }) =>
      dispatch(actions.pasteCell(payload)),
    createCellBelow: (payload: {
      id?: string | undefined;
      cellType: CellType;
      source: string;
      contentRef: string;
    }) => dispatch(actions.createCellBelow(payload)),
    changeCellType: (payload: {
      id?: string | undefined;
      to: CellType;
      contentRef: string;
    }) => dispatch(actions.changeCellType(payload)),
    openAboutModal: () =>
      dispatch(actions.openModal({ modalType: MODAL_TYPES.ABOUT })),
    changeKernelByName: (payload: {
      kernelSpecName: any;
      oldKernelRef?: string | null;
      contentRef: string;
    }) => dispatch(actions.changeKernelByName(payload)),
    restartKernel: (payload: {
      outputHandling: actions.RestartKernelOutputHandling;
      kernelRef?: string | null;
      contentRef: string;
    }) => dispatch(actions.restartKernel(payload)),
    restartKernelAndClearOutputs: (payload: {
      kernelRef?: string | null;
      contentRef: string;
    }) =>
      dispatch(
        actions.restartKernel({ ...payload, outputHandling: "Clear All" })
      ),
    restartKernelAndRunAllOutputs: (payload: {
      kernelRef?: string | null;
      contentRef: string;
    }) =>
      dispatch(
        actions.restartKernel({ ...payload, outputHandling: "Run All" })
      ),
    killKernel: (payload: { restarting: boolean; kernelRef?: string | null }) =>
      dispatch(actions.killKernel(payload)),
    interruptKernel: (payload: {
      kernelRef?: KernelRef | null;
      contentRef?: ContentRef | null;
    }) => dispatch(actions.interruptKernel(payload))
  });
  return mapDispatchToProps;
}

export const NotebookMenu = connect(
  makeMapStateToProps,
  makeMapDispatchToProps
)(PureNotebookMenu);

// We export this for testing purposes.
export { PureNotebookMenu };

export default NotebookMenu;
