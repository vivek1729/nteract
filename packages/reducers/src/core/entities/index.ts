// Vendor modules
import { makeEntitiesRecord } from "@nteract/types";
import { combineReducers } from "redux-immutable";

// Local modules
import { comms } from "./comms";
import { contents } from "./contents";
import { hosts } from "./hosts";
import { kernels } from "./kernels";
import { kernelspecs } from "./kernelspecs";
import { messages } from "./messages";
import { modals } from "./modals";
import { sidebar } from "./sidebar";
import { transforms } from "./transforms";

export const entities = combineReducers(
  {
    comms,
    contents,
    hosts,
    kernels,
    kernelspecs,
    messages,
    modals,
    sidebar,
    transforms,
  },
  makeEntitiesRecord as any
);
