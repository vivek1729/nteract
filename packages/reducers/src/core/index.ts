// Vendor modules
import * as actions from "@nteract/actions";
import { makeStateRecord } from "@nteract/types";
import { Action } from "redux";
import { combineReducers } from "redux-immutable";

// Local modules
import { entities } from "./entities";

// TODO: This can be removed once the work to rely on deriving KernelRefs
// ContentRefs is complete.
export const kernelRef = (state = "", action: Action) => {
  let typedAction;
  switch (action.type) {
    case actions.LAUNCH_KERNEL:
    case actions.LAUNCH_KERNEL_BY_NAME:
      typedAction = action as actions.LaunchKernelAction;
      return typedAction.payload.selectNextKernel
        ? typedAction.payload.kernelRef
        : state;
    case actions.LAUNCH_KERNEL_SUCCESSFUL:
      typedAction = action as actions.NewKernelAction;
      return typedAction.payload.selectNextKernel
        ? typedAction.payload.kernelRef
        : state;
    default:
      return state;
  }
};

export const currentKernelspecsRef = (state = "", action: Action) => {
  switch (action.type) {
    case actions.FETCH_KERNELSPECS:
      const typedAction = action as actions.FetchKernelspecs;
      return typedAction.payload.kernelspecsRef;
    default:
      return state;
  }
};

const core = combineReducers(
  {
    currentKernelspecsRef,
    entities,
    kernelRef
  },
  makeStateRecord as any
);

export default core;
