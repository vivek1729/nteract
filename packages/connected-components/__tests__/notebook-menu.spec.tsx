import { mount, shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import { mockAppState } from "@nteract/fixtures";

import { actions } from "@nteract/core";
import {
  makeMapDispatchToProps,
  makeMapStateToProps,
  PureNotebookMenu
} from "../src/notebook-menu";
import { MENU_ITEM_LABELS } from "../src/notebook-menu/constants";

describe("PureNotebookMenu", () => {
  describe("snapshots", () => {
    test("renders the default", () => {
      const component = renderer.create(<PureNotebookMenu />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("shallow", () => {
    test("renders the default", () => {
      const wrapper = shallow(<PureNotebookMenu />);
      expect(wrapper).not.toBeNull();
    });
  });
  describe("mount", () => {
    test("renders the default", () => {
      const wrapper = mount(<PureNotebookMenu />);
      expect(wrapper).not.toBeNull();
    });
    test("calls appropriate handlers on click", () => {
      const {
        DOWNLOAD_NOTEBOOK,
        EXECUTE_ALL_CELLS,
        EXECUTE_ALL_CELLS_BELOW,
        UNHIDE_ALL,
        CLEAR_ALL_OUTPUTS,
        CUT_CELL,
        COPY_CELL,
        PASTE_CELL,
        CREATE_MARKDOWN_CELL,
        CREATE_CODE_CELL,
        SET_CELL_TYPE_CODE,
        SET_CELL_TYPE_MARKDOWN,
        SET_THEME_LIGHT,
        SET_THEME_DARK,
        SAVE_NOTEBOOK,
        OPEN_ABOUT,
        INTERRUPT_KERNEL,
        RESTART_KERNEL,
        RESTART_AND_CLEAR_OUTPUTS,
        RESTART_AND_RUN_ALL_OUTPUTS,
        KILL_KERNEL
      } = MENU_ITEM_LABELS;
      const props = {
        // action functions
        executeCell: jest.fn(),
        executeAllCells: jest.fn(),
        executeAllCellsBelow: jest.fn(),
        unhideAll: jest.fn(),
        clearAllOutputs: jest.fn(),
        cutCell: jest.fn(),
        copyCell: jest.fn(),
        pasteCell: jest.fn(),
        createCellBelow: jest.fn(),
        changeCellType: jest.fn(),
        setTheme: jest.fn(),
        saveNotebook: jest.fn(),
        openAboutModal: jest.fn(),
        interruptKernel: jest.fn(),
        restartKernel: jest.fn(),
        restartKernelAndClearOutputs: jest.fn(),
        restartKernelAndRunAllOutputs: jest.fn(),
        killKernel: jest.fn(),
        downloadNotebook: jest.fn(),

        // document state (we mock out the implementation, so these are just
        // dummy variables.
        currentContentRef: "fake-content-ref",
        currentKernelRef: "fake-kernel-ref"
      };
      const wrapper = shallow(<PureNotebookMenu {...props} />);
      const eventMock = { preventDefault: () => null };

      const downloadNotebookItem = wrapper.find({ text: DOWNLOAD_NOTEBOOK });
      expect(props.downloadNotebook).not.toHaveBeenCalled();
      downloadNotebookItem.simulate("click", eventMock);
      expect(props.downloadNotebook).toHaveBeenCalledTimes(1);
      expect(props.downloadNotebook).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const executeAllCellsItem = wrapper.find({ text: EXECUTE_ALL_CELLS });
      expect(props.executeAllCells).not.toHaveBeenCalled();
      executeAllCellsItem.simulate("click", eventMock);
      expect(props.executeAllCells).toHaveBeenCalledTimes(1);
      expect(props.executeAllCells).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const executeAllCellsBelowItem = wrapper.find({
        text: EXECUTE_ALL_CELLS_BELOW
      });
      expect(props.executeAllCellsBelow).not.toHaveBeenCalled();
      executeAllCellsBelowItem.simulate("click", eventMock);
      expect(props.executeAllCellsBelow).toHaveBeenCalledTimes(1);
      expect(props.executeAllCellsBelow).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const unhideAllItem = wrapper.find({ text: UNHIDE_ALL });
      expect(props.unhideAll).not.toHaveBeenCalled();
      unhideAllItem.simulate("click", eventMock);
      expect(props.unhideAll).toHaveBeenCalledTimes(1);
      expect(props.unhideAll).toHaveBeenCalledWith({
        outputHidden: false,
        inputHidden: false,
        contentRef: props.currentContentRef
      });

      const clearAllOutputsItem = wrapper.find({ text: CLEAR_ALL_OUTPUTS });
      expect(props.clearAllOutputs).not.toHaveBeenCalled();
      clearAllOutputsItem.simulate("click", eventMock);
      expect(props.clearAllOutputs).toHaveBeenCalledTimes(1);
      expect(props.clearAllOutputs).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const cutCellItem = wrapper.find({ text: CUT_CELL });
      expect(props.cutCell).not.toHaveBeenCalled();
      cutCellItem.simulate("click", eventMock);
      expect(props.cutCell).toHaveBeenCalledTimes(1);
      expect(props.cutCell).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const copyCellItem = wrapper.find({ text: COPY_CELL });
      expect(props.copyCell).not.toHaveBeenCalled();
      copyCellItem.simulate("click", eventMock);
      expect(props.copyCell).toHaveBeenCalledTimes(1);
      expect(props.copyCell).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const pasteCellItem = wrapper.find({ text: PASTE_CELL });
      expect(props.pasteCell).not.toHaveBeenCalled();
      pasteCellItem.simulate("click", eventMock);
      expect(props.pasteCell).toHaveBeenCalledTimes(1);
      expect(props.pasteCell).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const createMarkdownCellItem = wrapper.find({
        text: CREATE_MARKDOWN_CELL
      });
      expect(props.createCellBelow).not.toHaveBeenCalled();
      createMarkdownCellItem.simulate("click", eventMock);
      expect(props.createCellBelow).toHaveBeenCalledTimes(1);
      expect(props.createCellBelow).toHaveBeenCalledWith({
        cellType: "markdown",
        contentRef: props.currentContentRef,
        source: ""
      });

      props.createCellBelow.mockClear();
      const createCodeCellItem = wrapper.find({ text: CREATE_CODE_CELL });
      expect(props.createCellBelow).not.toHaveBeenCalled();
      createCodeCellItem.simulate("click", eventMock);
      expect(props.createCellBelow).toHaveBeenCalledTimes(1);
      expect(props.createCellBelow).toHaveBeenCalledWith({
        cellType: "code",
        contentRef: props.currentContentRef,
        source: ""
      });

      const setCellTypeCodeItem = wrapper.find({ text: SET_CELL_TYPE_CODE });
      expect(props.changeCellType).not.toHaveBeenCalled();
      setCellTypeCodeItem.simulate("click", eventMock);
      expect(props.changeCellType).toHaveBeenCalledTimes(1);
      expect(props.changeCellType).toHaveBeenCalledWith({
        to: "code",
        contentRef: props.currentContentRef
      });

      props.changeCellType.mockClear();
      const setCellTypeMarkdownItem = wrapper.find({
        text: SET_CELL_TYPE_MARKDOWN
      });
      expect(props.changeCellType).not.toHaveBeenCalled();
      setCellTypeMarkdownItem.simulate("click", eventMock);
      expect(props.changeCellType).toHaveBeenCalledTimes(1);
      expect(props.changeCellType).toHaveBeenCalledWith({
        to: "markdown",
        contentRef: props.currentContentRef
      });

      const setThemeLightItem = wrapper.find({ text: SET_THEME_LIGHT });
      expect(props.setTheme).not.toHaveBeenCalled();
      setThemeLightItem.simulate("click", eventMock);
      expect(props.setTheme).toHaveBeenCalledTimes(1);
      expect(props.setTheme).toHaveBeenCalledWith("light");

      props.setTheme.mockClear();
      const setThemeDarkItem = wrapper.find({ text: SET_THEME_DARK });
      expect(props.setTheme).not.toHaveBeenCalled();
      setThemeDarkItem.simulate("click", eventMock);
      expect(props.setTheme).toHaveBeenCalledTimes(1);
      expect(props.setTheme).toHaveBeenCalledWith("dark");

      const saveNotebookItem = wrapper.find({ text: SAVE_NOTEBOOK });
      expect(props.saveNotebook).not.toHaveBeenCalled();
      saveNotebookItem.simulate("click", eventMock);
      expect(props.saveNotebook).toHaveBeenCalledTimes(1);
      expect(props.saveNotebook).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const openAboutItem = wrapper.find({ text: OPEN_ABOUT });
      expect(props.openAboutModal).not.toHaveBeenCalled();
      openAboutItem.simulate("click", eventMock);
      expect(props.openAboutModal).toHaveBeenCalledTimes(1);
      expect(props.openAboutModal).toHaveBeenCalledWith();

      const interruptKernelItem = wrapper.find({ text: INTERRUPT_KERNEL });
      expect(props.interruptKernel).not.toHaveBeenCalled();
      interruptKernelItem.simulate("click", eventMock);
      expect(props.interruptKernel).toHaveBeenCalledTimes(1);
      expect(props.interruptKernel).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const restartKernelItem = wrapper.find({ text: RESTART_KERNEL });
      expect(props.restartKernel).not.toHaveBeenCalled();
      restartKernelItem.simulate("click", eventMock);
      expect(props.restartKernel).toHaveBeenCalledTimes(1);
      expect(props.restartKernel).toHaveBeenCalledWith({
        outputHandling: "None",
        contentRef: props.currentContentRef
      });

      const restartKernelAndClearOutputsItem = wrapper.find({
        text: RESTART_AND_CLEAR_OUTPUTS
      });
      expect(props.restartKernelAndClearOutputs).not.toHaveBeenCalled();
      restartKernelAndClearOutputsItem.simulate("click", eventMock);
      expect(props.restartKernelAndClearOutputs).toHaveBeenCalledTimes(1);
      expect(props.restartKernelAndClearOutputs).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const restartKernelAndRunAllOutputsItem = wrapper.find({
        text: RESTART_AND_RUN_ALL_OUTPUTS
      });
      expect(props.restartKernelAndRunAllOutputs).not.toHaveBeenCalled();
      restartKernelAndRunAllOutputsItem.simulate("click", eventMock);
      expect(props.restartKernelAndRunAllOutputs).toHaveBeenCalledTimes(1);
      expect(props.restartKernelAndRunAllOutputs).toHaveBeenCalledWith({
        contentRef: props.currentContentRef
      });

      const killKernelItem = wrapper.find({ text: KILL_KERNEL });
      expect(props.killKernel).not.toHaveBeenCalled();
      killKernelItem.simulate("click", eventMock);
      expect(props.killKernel).toHaveBeenCalledTimes(1);
      expect(props.killKernel).toHaveBeenCalledWith({
        kernelRef: props.currentKernelRef,
        contentRef: props.currentContentRef,
        restarting: false
      });
    });
  });
});

describe("makeMapStateToProps", () => {
  it("returns default values if content is not a notebook", () => {
    const state = mockAppState({});
    const result = makeMapStateToProps(state, { contentRef: "test" })(state);
    expect(result).toEqual({
      currentContentRef: "test",
      currentKernelRef: null,
      currentKernelspecs: null,
      currentKernelspecsRef: null
    });
  });
  it("returns default values if content isa notebook", () => {
    const state = mockAppState({});
    const contentRef = state.core.entities.contents.byRef.keySeq().first();
    const result = makeMapStateToProps(state, { contentRef })(state);
    expect(result).toEqual({
      currentContentRef: contentRef,
      currentKernelRef: expect.any(String),
      currentKernelspecs: null,
      currentKernelspecsRef: null,
      bookstoreEnabled: false
    });
  });
});

describe("makeMapDispatchToProps", () => {
  it("returns the required actions", () => {
    const dispatch = jest.fn();
    const result = makeMapDispatchToProps(dispatch, { contentRef: "test " })(
      dispatch
    );
    result.saveNotebook({ contentRef: "test" });
    expect(dispatch).toBeCalledWith(actions.save({ contentRef: "test" }));
  });
});
