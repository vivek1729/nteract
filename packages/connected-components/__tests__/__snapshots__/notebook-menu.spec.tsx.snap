// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureNotebookMenu snapshots renders the default 1`] = `
<ul
  className="bp3-menu"
  style={
    Object {
      "background": "var(--theme-app-bg)",
      "borderBottom": "1px solid var(--theme-app-border)",
      "color": "var(--theme-app-fg)",
      "display": "inline-flex",
      "position": "sticky",
      "top": 0,
      "width": "100%",
      "zIndex": 19,
    }
  }
>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            File
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            Edit
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            View
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            Cell
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            Runtime
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
  <li
    className="bp3-submenu"
  >
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onClick={[Function]}
      >
        <a
          className="bp3-menu-item"
        >
          <div
            className="bp3-text-overflow-ellipsis bp3-fill"
          >
            Help
          </div>
          <span
            className="bp3-icon bp3-icon-caret-right"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                caret-right
              </desc>
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </span>
  </li>
</ul>
`;
